### ISSUE TITLE
How to get confidence of the predicted point

### ISSUE THREAD

**Issue by srinivas1746** [ashwhall/dsnt#14]
I am using DSNT to predict the coordinate, is there any way that I can get the confidence of the predicted point.

**Comment by ashwhall** [comment]
Hi @srinivas1746. DSNT doesn't produce a true confidence prediction, as there is nothing during training to enforce it.

The network produces an unnormalised prediction, which is then rectified using the softmax function. The magnitude of the unnormalised outputs _may_ be considered some form of confidence, but as it's unnormalised the values are near impossible to interpret.
The softmax function will always sum to 1 across the spatial dimensions, so you're effectively given the confidence of each pixel coordinate, but this doesn't provide the overall confidence for the existence of the keypoint itself.

In summary, no there isn't a way.
